EXAMPLES.md

```diff
@@ -1234,6 +1234,148 @@ Once it's applied, whoami should be accessible on https://whoami.docker.localhos
+
+```bash
+# Install NGINX Ingress Controller
+helm upgrade --install ingress-nginx ingress-nginx \
+  --repo https://kubernetes.github.io/ingress-nginx \
+  --namespace ingress-nginx --create-namespace
+```
+
+Deploy the application:
+
+```yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: whoami
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: whoami
+  template:
+    metadata:
+      labels:
+        app: whoami
+    spec:
+      containers:
+        - name: whoami
+          image: traefik/whoami
+          ports:
+            - containerPort: 80
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: whoami
+spec:
+  selector:
+    app: whoami
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 80
+
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: whoami
+  annotations:
+    nginx.ingress.kubernetes.io/affinity: cookie
+    nginx.ingress.kubernetes.io/affinity-mode: persistent 
+spec:
+  ingressClassName: nginx
+  rules:
+    - host: whoami.docker.localhost
+      http:
+        paths:
+          - path: /
+            pathType: Prefix
+            backend:
+              service:
+                name: whoami
+                port:
+                  number: 80
+```
+
+**Step 2**: Test that the application works with NGINX
+
+```bash
+# Port-forward to NGINX
+kubectl port-forward -n ingress-nginx deployment/ingress-nginx-controller 8000:80 &
+
+# Test with NGINX
+curl http://whoami.docker.localhost:8000 -c /tmp/cookies.txt -b /tmp/cookies.txt
+```
+
+You should see the whoami response with your request details.
+
+**Step 3**: Install Traefik with Kubernetes Ingress NGINX provider enabled (alongside NGINX)
+
+```bash
+helm upgrade --install traefik traefik/traefik \
+  --namespace traefik --create-namespace \
+  --set providers.kubernetesIngressNginx.enabled=true
+```
+
+Or using a values file:
+
+```yaml
+providers:
+  kubernetesIngressNginx:
+    enabled: true
+```
+
+**Step 4**: Test that the application now also works with Traefik
+
+Both NGINX and Traefik are now running in parallel, each serving the same Ingress thanks to the Traefik NGINX provider !
+
+```bash
+# Port-forward to Traefik
+kubectl port-forward -n traefik deployment/traefik 8001:8000 &
+
+# Test with Traefik (adjust the URL based on your setup)
+curl http://whoami.docker.localhost:8001 -c /tmp/cookies.txt -b /tmp/cookies.txt
+```
+
+The same Ingress resource is now served by **both** NGINX and Traefik! You can verify which one is responding by checking the response headers or the service endpoints.
+
+> [!WARNING]
+> **Important note about uninstalling NGINX**: When you uninstall the NGINX Ingress Controller helm chart, it will remove the `nginx` IngressClass.
+> Traefik needs this IngressClass to detect and serve Ingress resources that use `ingressClassName: nginx`. If you uninstall NGINX, you'll need to manually create the IngressClass like this:
```

Comment:
```suggestion
> Traefik needs this IngressClass to detect and serve Ingress resources that use `ingressClassName: nginx`. Before uninstalling NGINX, it's recommended to ensure that an IngressClass like this will stay:
```